src/smc-webapp/projects.cjsx

```diff
@@ -231,7 +231,10 @@ class ProjectsActions extends Actions
         if opts.target?
             redux.getProjectActions(opts.project_id)?.load_target(opts.target, opts.switch_to)
         redux.getActions('page').save_session()
-        project_actions.check_library()
+        # init the library after project started.
+        # TODO write a generalized store function that does this in a more robust way
```

Comment:
Hmm, I think you started to look at older commits. I'm not sure how all this will end up. My thoughts here where that the parsed json data should also be validated in some way. Here, the only test is if a file exists on the disk.